### PR TITLE
Use flake check for the documentation linting test

### PR DESCRIPTION
**Pull request by nlewo**
The goal is to easily run the documentation CI tests locally. 

- It moves the documentation linting test as a `flake check` attribute.
- The script `lint-docs` is now part of the shell environment.

The idea would be to move more CI scripts to `nix flake check`: it is easier to discover them and to reproduce them.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -7,6 +7,28 @@
 
   outputs = { self, nixpkgs, utils }: utils.lib.eachDefaultSystem (system: let
     pkgs = import nixpkgs { inherit system; };
+    # TODO: need to be moved as flake input once this repository is a flake
+    gitignore = import (builtins.fetchTarball {
```
Comment (FRidh) [inline flake.nix:11]: I don't think we can use `fetchTarball` on Hydra.

File: flake.nix [hunk 2]
```diff
@@ -7,6 +7,28 @@
...
+      url = https://github.com/hercules-ci/gitignore/archive/647d0821b590ee96056f4593640534542d8700e5.tar.gz;
+      sha256 = "0ks37vclz2jww9q0fvkk9jyhscw0ial8yx2fpakra994dm12yy1d";
+    }) { inherit (pkgs) lib; };
+
+    sphinx = pkgs.python3.withPackages(p: [ p.sphinx ]);
```
Comment (adisbladis) [inline flake.nix:16]: Use the environment provided by `mkPoetryEnv`, we don't want to be using multiple different versions of `sphinx`.
https://github.com/NixOS/nixops/blob/master/flake.nix#L13-L15

File: flake.nix (hunk 1 above)
Comment (nlewo) [reply flake.nix:11]: This is correct (excepting if `allowed_uris` is set). 

I've planned to send a PR to the `hercules-ci/gitignore` to add the flake support.

Is it a short term issue? 
In this case, we can wait until the `gitignore` flake support is merged to merge this PR: I could then put this lib as a flake input.

Comment (nlewo) [reply flake.nix]: I finally switch to the `poetry2nix.cleanPythonSources` which works with the restricted evaluation.

**Comment by nlewo** [comment]
@adisbladis I think I've addressed all requests. Could you take a look?

**Pull request merged**